### ISSUE TITLE
ModelVersionWarning 

### ISSUE THREAD

**Issue by koaning** [RasaHQ/rasa#5699]
The bug described [here](https://github.com/RasaHQ/rasa-x/issues/2573) was fixed after removing models that were trained with an old version of rasa. The error message did not at all suggest this. Meantioning @erohmensing because she was also looking at it. 

```
  File "/Users/vincent/Development/pokebot/venv/lib/python3.6/site-packages/rasa/nlu/classifiers/diet_classifier.py", line 1373, in batch_loss
    sequence_lengths = self.sequence_lengths_for(tf_batch_data[TEXT_SEQ_LENGTH][0])
```

These were my version numbers:

```
OS-X python3.6
rasa==1.9.6
rasa-core-sdk==0.14.0
rasa-sdk==1.9.0
rasa-x==0.27.6
```

It feels like we might be able to catch this is a more friendly way. If a model was trained on a version of rasa that is not currently running then we can warn the user of this explicitly.

**Comment by erohmensing** [comment]
@tabergma so do we need to update the minimum compatible version on 1.9.x then? (@koaning for reference we do have code that does `If a model was trained on a version of rasa that is not currently running then we can warn the user of this explicitly.`)

**Comment by tabergma** [comment]
Yes, we updated something I think in 1.9.5 that was actually model breaking. We did not increased the model version back then, because we forgot about it. We should not have merged this PR into 1.9.x but into master. Sorry, about that. We should increase the model version.

**Comment by erohmensing** [comment]
Just released 1.9.7 reverting the changes, which is compatible with all 1.9 models (except 1.9.6) -- Think this can be closed
